### PR TITLE
Prevent stale broker load data in ConfigMap after proposal refresh

### PR DESCRIPTION
**Pull request by kyguy**
### Type of change

- Bugfix

### Description

When a `KafkaRebalance` custom resource in `Ready` or `ProposalReady` state is refreshed using the `refresh` annotation, the KafkaRebalanceOperator populates the Cruise Control `ConfigMap` with stale broker load information from the previous proposal instead of the new (refreshed) proposal.

The offending code was intended to preserve broker load information between progress updates of a `KafkaRebalance` resource in the "Rebalancing" state. However,  the code is not needed since the KafkaRebalanceOperator already copies ConfigMap data from previous states when creating the `MapAndStatus<ConfigMap, KafkaRebalanceStatus>` object for new state transitions. 

This PR removes the offending code and adds test coverage for this edge case and other state transitions involving broker load retention and refresh.

Credit to @tomncooper @MikeEdgar for finding the bug!

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update documentation
- [ ] Update CHANGELOG.md (if present)
- [ ] Reference relevant issue(s) and close them after merging
- [x] Write tests
- [ ] Make sure all tests pass
- [x] Try your changes inside a Kubernetes cluster, not just from unit tests
- [x] AI assistance was used to create this PR (see the [Strimzi AI policy](https://github.com/strimzi/governance/blob/main/AI_POLICY.md))

### CODE REVIEW

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperatorProgressTest.java [hunk 1]
```diff
@@ -116,6 +120,24 @@ private Future<Void> verifyKafkaRebalanceStateAndConfigMap(VertxTestContext cont
...
+        return Future.succeededFuture(current);
+    }
+
+    private Future<ConfigMap> verifyBrokerLoadChanged(VertxTestContext context, KafkaRebalanceState state, ConfigMap previous, ConfigMap current) {
+        context.verify(() -> {
+            assertState(context, client, namespace, RESOURCE_NAME, state);
+            assertThat(current.getData().get(BROKER_LOAD_KEY), is(not(previous.getData().get(BROKER_LOAD_KEY))));
+        });
+        return Future.succeededFuture(current);
+    }
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperatorProgressTest.java]: These methods seem weird. Why do you need to do the assert asynchronously? If the asserts are repeated, having them in a separate method seems reasonable. But the async handling should be IMHO inlined. You could also use something like `onComplete()` instead of `compose` for it.

Comment (kyguy) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperatorProgressTest.java]: Yeah you are right, the async logic is not necessary here. I have just refactored to use inlined `onSuccess()` statements from the calling method instead in the latest commit, let me know what you think!

**Review by scholzj — APPROVED** [review]

**Review by ShubhamRwt — APPROVED** [review]
LGTM

**Review by ppatierno — APPROVED** [review]
